cmd/restic/cmd_ls.go

```diff
@@ -59,20 +74,89 @@ func runLs(opts LsOptions, gopts GlobalOptions, args []string) error {
 		return err
 	}
 
+	// extract any specific directories to walk
```

Comment:
I'd do that before opening the repo, so users get a feedback on the dirs they specified (e.g. when they aren't absolute) early on, before any network traffic.